### PR TITLE
feat: update walletconnect v2

### PR DESCRIPTION
**Pull request by namgold**

### CODE REVIEW

File: src/components/Header/web3/SelectWallet.tsx [hunk 1]
```diff
@@ -121,7 +119,16 @@ function Web3StatusInner() {
...
+  if (isWrongNetwork) {
+    return (
+      <Web3StatusError onClick={toggleNetworkModal}>
+        <NetworkIcon />
+        <Text>
+          <Trans>Wrong Network</Trans>
+        </Text>
+      </Web3StatusError>
+    )
+  } else if (account) {
```
Comment (nguyenhoaidanh) [inline src/components/Header/web3/SelectWallet.tsx:131]: rm `else`

File: src/connectors/index.ts [hunk 2]
```diff
@@ -1,56 +1,57 @@
...
+        }, {} as { [key in ChainId]: string }),
+        qrModalOptions: {
+          themeMode: darkmode ? 'dark' : 'light',
+          themeVariables: {
+            '--w3m-z-index': '1000',
+          },
+        },
+        metadata: {
+          name: 'Kyberswap',
+          description: 'Kyberswap - Trading smart',
```
Comment (nguyenhoaidanh) [inline src/connectors/index.ts:40]: description: document.title

File: src/connectors/utils.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
+// import { Connection, ConnectionType } from 'connection/types'
+import { Connector } from '@web3-react/types'
+
+import { coinbaseWallet, walletConnectV2 } from 'connectors'
+
+export const getIsInjected = () => Boolean(window.ethereum)
+
+type NonMetaMaskFlag = 'isRabby' | 'isBraveWallet' | 'isTrustWallet' | 'isLedgerConnect'
+const allNonMetamaskFlags: NonMetaMaskFlag[] = ['isRabby', 'isBraveWallet', 'isTrustWallet', 'isLedgerConnect']
```
Comment (nguyenhoaidanh) [inline src/connectors/utils.ts:9]: or
```const allNonMetamaskFlags = ['isRabby', 'isBraveWallet', 'isTrustWallet', 'isLedgerConnect'] as const```

File: src/constants/networks.ts [hunk 4]
```diff
@@ -111,6 +111,10 @@ export function isSolana(chainId?: ChainId): chainId is ChainId.SOLANA {
   const chainType = getChainType(chainId)
   return chainType === ChainType.SOLANA
 }
+export function isSupportedChainId(chainId?: number): chainId is ChainId {
```
Comment (nguyenhoaidanh) [inline src/constants/networks.ts:114]:
```
export function isSupportedChainId(chainId?: number) {
  if (!chainId) return false
  return !!NETWORKS_INFO_CONFIG[chainId as ChainId]
}
```

File: src/hooks/useSyncNetworkParamWithStore.ts [hunk 5]
```diff
@@ -70,8 +35,7 @@ export function useSyncNetworkParamWithStore() {
           isOnInit.current = false
           return
         }
-        if ((possibleWallet || possibleWallet === null) && !account) {
-          // connected wallet but web3-react slow return account, or checking connection
+        if (!account) {
```
Comment (nguyenhoaidanh) [inline src/hooks/useSyncNetworkParamWithStore.ts:38]: active network is eth
enter https://kyberswap-interface-1954.pr.kyberengineering.io/swap/bnb => not change change => bug

File: src/connectors/utils.ts (hunk 3 above)
Comment (namgold) [reply src/connectors/utils.ts:9]: This file is copied from uniswap 🤫

File: src/hooks/useSyncNetworkParamWithStore.ts (hunk 5 above)
Comment (nguyenhoaidanh) [reply src/hooks/useSyncNetworkParamWithStore.ts:38]: remove if (!account)

File: src/constants/wallets.ts [hunk 6]
```diff
@@ -19,70 +12,85 @@ import SLOPE from 'assets/wallets-connect/slope.svg'
...
-  if (window.coinbaseWalletExtension) return WalletReadyState.Loadable
-  return WalletReadyState.NotDetected
+  const detectCoinbase = (): WalletReadyState => {
+    if (getIsCoinbaseWallet()) return WalletReadyState.Installed
+    // in NotDetected case, Coinbase show install link itself
+    if (window.coinbaseWalletExtension && !isMobile) return WalletReadyState.Loadable
+    return WalletReadyState.NotDetected
+  }
+  const result = detectCoinbase()
+  // alert(`detectCoinbase ${result}`)
```
Comment (nguyenhoaidanh) [inline src/constants/wallets.ts:80]: rm

**Review by nguyenhoaidanh — DISMISSED** [review]
LGTM, wait for @hungdoansy to complete his review

File: src/hooks/useSyncNetworkParamWithStore.ts (hunk 5 above)
Comment (namgold) [reply src/hooks/useSyncNetworkParamWithStore.ts]: fixed

**Review by nguyenhoaidanh — APPROVED** [review]

**Pull request merged**